pkg/loki/modules.go

```diff
@@ -1081,6 +1081,11 @@ func (t *Loki) initCompactor() (services.Service, error) {
 	// Set some config sections from other config sections in the config struct
 	t.Cfg.CompactorConfig.CompactorRing.ListenPort = t.Cfg.Server.GRPCListenPort
 
+	if !(config.UsingObjectStorageIndex(t.Cfg.SchemaConfig.Configs) || config.UsingBoltDBtorageIndex(t.Cfg.SchemaConfig.Configs)) {
```

Comment:
This is concerning. Why are we turning off index compaction when using the filesystem? It still provides a performance benefit, even if the chunks are local.